### PR TITLE
#632: Indicate `Rust` or `(rust)` for rust_version segment only if ic…

### PR DESCRIPTION
**Pull request by chhschou**
…on is unavailable

Hi @bhilburn, please let me know what you think.

### CODE REVIEW

**Comment by bhilburn** [comment]
Thanks, @chhschou! In looking at the code, here, you're right - this needs to get fixed.

Your PR looks good, but in looking at the code, it looks like there is even more we could fix. So, check this out:

* `awesome-patched`: You are fixing this with with `(rust`) [here](https://github.com/bhilburn/powerlevel9k/pull/641/files#diff-b6c6e7997da6b6ad40e6e917acafa27cR80)
* `awesome-fontconfig`: Mapped to `E6A8` [here](https://github.com/bhilburn/powerlevel9k/pull/641/files#diff-b6c6e7997da6b6ad40e6e917acafa27cL151)
* `nerdfont-complete`: Mapped to `E7A8` [here](https://github.com/bhilburn/powerlevel9k/pull/641/files#diff-b6c6e7997da6b6ad40e6e917acafa27cL222)
* `powerline-patched`: You fix this with `Rust` [here](https://github.com/bhilburn/powerlevel9k/pull/641/files#diff-b6c6e7997da6b6ad40e6e917acafa27cR293)

I have a couple of questions:

1) Is there a reason you used `(rust)` in one place and `Rust` in the other?
2) For the most part, `awesome-patched` and `awesome-fontconfig` can use the same codepoints. Are we sure that these need to be different?
3) So, I pulled up the `awesome-fontconfig` font map, and guess what? `E6A8` isn't anywhere in there. [Check it out](https://github.com/gabrielelana/awesome-terminal-fonts/blob/master/fonts/fontawesome-regular.sh) -- can you comment, @gabrielelana?
4) It's also strange to me that the Nerdfonts codepoint is only just slightly different from the Awesome map, `E6A8` vs `E7A8`. -- can you comment, @ryanoasis?

**Comment by chhschou** [comment]
@bhilburn I have a pretty basic understanding of `awesome-patched` and `powerline-patched`. I used both `(rust)` and `Rust` following the patterns surrounding each change. If there is a preference for one over the other I'm happy to comply. I did ask a question in #632 but it wasn't answered so I thought the choice wasn't important. Personally, I like `(rust)` better, the parentheses helps me see that the icon is unavailable.

**Comment by gabrielelana** [comment]
@bhilburn 

> So, I pulled up the awesome-fontconfig font map, and guess what? E6A8 isn't anywhere in there. Check it out -- can you comment, @gabrielelana?

Because the rust icon is not part of the [fontawesome](http://fontawesome.io/icons/) icon font but is part of [devicons](http://vorillaz.github.io/devicons/#/dafont) icon font and in fact you can find the font code point in the [devicon font map](https://github.com/gabrielelana/awesome-terminal-fonts/blob/master/fonts/devicons-regular.sh)

> For the most part, awesome-patched and awesome-fontconfig can use the same codepoints. Are we sure that these need to be different?

for `awesome-fontconfig` always use the variables defined in the [font map](https://github.com/gabrielelana/awesome-terminal-fonts/blob/master/fonts/devicons-regular.sh), so use `"\u$CODEPOINT_OF_DEVICONS_RUST"` instead of `$'\uE6A8'` if you do that then I can update the icon fonts, possibly shift icons around to avoid clashes and still don't break anything for anyone

**Comment by ryanoasis** [comment]
@bhilburn 

> It's also strange to me that the Nerdfonts codepoint is only just slightly different from the Awesome map, E6A8 vs E7A8

As @gabrielelana has said that glyph is from [devicons](https://vorillaz.github.io/devicons/#/cheat) not font awesome. Nerd Fonts has shifted the codepoints range due to conflicts with other glyph fonts: `E600-E6C5 → E700-E7C5`.

It occurs to me the shifting isn't clear on the [readme](https://github.com/ryanoasis/nerd-fonts#devicons) of Nerd Fonts currently :disappointed: 

Also we somewhat recently added variables for mapping the codepoints as well, e.g. [devicons map](https://github.com/ryanoasis/nerd-fonts/blob/master/bin/scripts/lib/i_dev.sh) though honestly I am not sure how `P9k` would incorporate them.

File: functions/icons.zsh [hunk 1]
```diff
@@ -290,7 +290,7 @@ case $POWERLEVEL9K_MODE in
       VCS_GIT_GITLAB_ICON            ''
       VCS_HG_ICON                    ''
       VCS_SVN_ICON                   ''
-      RUST_ICON                      ''
+      RUST_ICON                      'Rust'
```
Comment (dritter) [inline functions/icons.zsh]: At some point we should agree on parenthesis or not.. But that is not the issue in this PR.

**Pull request merged**